services/ui-src/src/components/ComponentId.tsx

```diff
@@ -64,21 +70,25 @@ const ComponentId: React.FC<{
                 {message.statusMessage}
               </div>
             ))}
-          <input
-            className="field"
-            type="text"
-            id={idPrefix + "componentId"}
-            name="componentId"
-            aria-describedby={idFieldHint
-              ?.map(function (idFieldHint, idx) {
-                return idPrefix + "fieldHint" + idx;
-              })
-              .join(" ")}
-            value={value}
-            onChange={onChange}
-            required
-            disabled={disabled}
-          ></input>
+          <div className="all-id">
+            {prefill && <span className="prefill-id">{prefill}</span>}
+            <input
+              className={"field-" + idSize}
```

Comment:
The hesitance is because I struggled with figuring out the different sizes without using business logic... the Initial waiver medium field looked bad small and the TE ID field looked too big as medium and I suppose you can say "pre gets small and post gets medium" but that felt too much like business logic when it could be a setting at the config level (and actually part of the business logic), where we really do know the desired size. I also didn't want fields to change size as the prefills were typed.
What I *really* wanted was to get to deeper ID requirement sharing and the form merely states which part gets edited.  But that didn't work in the time I had, so this is where I landed.

Reply:
... I also struggle with typing and design in general, and fully admit I will not make the best choices in either, so the suggestions are great :)